File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        int lower = 0;
        int upper = 0;
        
        for (char c : word.toCharArray()) {
            if (c >= 'a' && c <= 'z') {
                lower |= (1 << (c - 'a'));
            } else {
                upper |= (1 << (c - 'A'));
            }
        }
        
        return Integer.bitCount(lower & upper);
    }
}

File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lower = 0
        upper = 0
        
        for char in word:
            if char.islower():
                lower |= 1 << (ord(char) - 97)
            else:
                upper |= 1 << (ord(char) - 65)
                
        return (lower & upper).bit_count()